src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -371,6 +371,11 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
             subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
             namespace='Microsoft.Compute', type='diskAccesses', name=disk_access)
 
+    if secure_vm_disk_encryption_set is not None and not is_valid_resource_id(secure_vm_disk_encryption_set):
+        secure_vm_disk_encryption_set = resource_id(
+            subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
+            namespace='Microsoft.Compute', type='diskEncryptionSets', name=secure_vm_disk_encryption_set)
```

Comment:
@AjKundnani Do we need to consider supporting `--secure-vm-disk-encryption-set` to pass in name in addition to ID (just like `--disk-encryption-set`)

Reply:
@zhoxing-ms - Yes please, if we can support resource name for consistency.

Reply:
@cxznmhdcxz Could you add some tests for passing in the name to `--secure-vm-disk-encryption-set`?